### PR TITLE
Activate full screen image

### PR DESCRIPTION
**Pull request by misspia**
## [DEMO](https://share.getcloudapp.com/YEu184nZ)

- Converts the FullscreenImageView component into a UIViewController
- Reveals the fullscreen image on battle preview tap
- Returns back to previous view controller on fullscreen image tap

### CODE REVIEW

File: SnapWar/Components/SWImagePreview.swift [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import UIKit
 
 class SWImagePreview: UIImageView {
-    var named: String
+    let named: String
+    var isActive: Bool = false
```
Comment (davidozhang) [inline SnapWar/Components/SWImagePreview.swift]: don't think this is used, can remove

File: SnapWar/ViewControllers/SWBattleViewController.swift [hunk 2]
```diff
@@ -1,51 +1,62 @@
...
         } else {
             previewContainer.axis = .vertical
         }
     }
+    @objc func handleTap(sender: UITapGestureRecognizer) {
+        if(sender.state != .ended) {
+            return
+        }
+        
+        let preview = sender.view as! SWImagePreview
```
Comment (davidozhang) [inline SnapWar/ViewControllers/SWBattleViewController.swift]: let's not use !, instead you should use `guard let ... else { throw some error }`

File: SnapWar/ViewControllers/SWBattleViewController.swift [hunk 3]
```diff
@@ -1,51 +1,62 @@
...
-        fullScreenImageView.handleResize()
-        
+
         if(UIWindow.isLandscape) {
              previewContainer.axis = .horizontal
         } else {
             previewContainer.axis = .vertical
         }
     }
+    @objc func handleTap(sender: UITapGestureRecognizer) {
```
Comment (davidozhang) [inline SnapWar/ViewControllers/SWBattleViewController.swift]: nit: add an extra line above this method declaration, should also put @objc on its own line

File: SnapWar/ViewControllers/SWFullScreenImageViewController.swift [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        imageView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor).isActive = true
+        
+        imageView.contentMode = .scaleAspectFit
+        
+        let tap = UITapGestureRecognizer(target: self, action: #selector(self.handleTap))
+        imageView.addGestureRecognizer(tap)
+        imageView.isUserInteractionEnabled = true
+    }
+    
+    @objc func handleTap(sender: UITapGestureRecognizer) {
```
Comment (davidozhang) [inline SnapWar/ViewControllers/SWFullScreenImageViewController.swift:42]: also nit: @objc on own line

**Review by davidozhang — APPROVED** [review]
LGTM, small comments 😄

**Comment by davidozhang** [comment]
the demo is super sick btw 💯

File: SnapWar/ViewControllers/SWFullScreenImageViewController.swift [hunk 5]
```diff
@@ -0,0 +1,49 @@
+
+import UIKit
+
+class SWFullscreenImageViewController: UIViewController {
+    var imageView: UIImageView
```
Comment (davidozhang) [inline SnapWar/ViewControllers/SWFullScreenImageViewController.swift:5]: fileprivate

File: SnapWar/ViewControllers/SWFullScreenImageViewController.swift [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view?.backgroundColor = SWColorType.black
+        setup()
+    }
+    
+    func setup() {
```
Comment (davidozhang) [inline SnapWar/ViewControllers/SWFullScreenImageViewController.swift:26]: fileprivate

File: SnapWar/ViewControllers/SWBattleViewController.swift [hunk 7]
```diff
@@ -1,51 +1,61 @@
...
+
         let testImageNames = [
             "test-image-1",
             "test-image-3",
         ]
         addPreviewImages(testImageNames)
     }
-    
+
     func addPreviewImages(_ imageNames: Array<String>) {
```
Comment (davidozhang) [inline SnapWar/ViewControllers/SWBattleViewController.swift:18]: fileprivate

File: SnapWar/ViewControllers/SWFullScreenImageViewController.swift [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        view.addSubview(imageView)
+
+        imageView.translatesAutoresizingMaskIntoConstraints = false
```
Comment (davidozhang) [inline SnapWar/ViewControllers/SWFullScreenImageViewController.swift:29]: im thinking we should have a swimageview that does this automatically so we don't have to replicate this layout code everytime

Comment (misspia) [reply SnapWar/ViewControllers/SWFullScreenImageViewController.swift:29]: ^ Will create this in a separate branch

**Pull request merged**